test/tests.rb

```diff
@@ -45,33 +47,29 @@ def log_level_test(level, port, expected_level)
   end
 
   # Should retry to connect and preserve the failed line
-  def fatal_method_not_found(level, port, expected_level)
+  def retry_test(level, port, _expected_level)
     second_line = " second line"
     options = get_options(port)
-    logdna_thread = Thread.start do
-      logger = Logdna::Ruby.new("pp", options)
-      logger.send(level, LOG_LINE)
-      logger.send(level, second_line)
-    end
+
+    logger = Logdna::Ruby.new("pp", options)
+    logger.send(level, LOG_LINE)
 
     server_thread = Thread.start do
-      sor = TestServer.new
-      recieved_data = sor.return_not_found_res(port)
-      # The order of recieved lines is unpredictable.
-      recieved_lines = [
-        recieved_data[:ls][0][:line],
-        recieved_data[:ls][1][:line]
-      ]
+      server_generator = TestServer.new
+      server = server_generator.start_server(port)
+      server_generator.accept_logs_and_respond(server, "HTTP/1.1 408 Request Timeout")
+      # make a second request
+      logger.send(level, second_line)
 
-      assert_includes(recieved_lines, LOG_LINE)
-      assert_includes(recieved_lines, second_line)
+      # Both lines will come back in separate requests
+      recieved_data_1 = server_generator.accept_logs_and_respond(server, "HTTP/1.1 200 OK")
+      recieved_data_2 = server_generator.accept_logs_and_respond(server, "HTTP/1.1 200 OK")
 
-      assert_equal(recieved_data[:ls][0][:app], options[:app])
-      assert_equal(recieved_data[:ls][0][:level], expected_level)
-      assert_equal(recieved_data[:ls][0][:env], options[:env])
+      # The order of recieved lines is unpredictable.
+      assert_includes([recieved_data_1[:ls][0][:line], recieved_data_2[:ls][0][:line]], LOG_LINE)
+      assert_includes([recieved_data_1[:ls][0][:line], recieved_data_2[:ls][0][:line]], second_line)
```

Comment:
why are the asserts here same, @jakedipity?

Reply:
Not a duplicate. Since it's non-deterministic we have to check both payloads for each line.